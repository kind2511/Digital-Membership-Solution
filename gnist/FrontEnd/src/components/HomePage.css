.home-page {
    display: flex;
    height: 100vh; /* Full height */
}

/* Style for the left side of the home page */
.home-page .left {
    width: 50%;
    background-color: #ffd700;
    display: flex;
    align-items: center;
    justify-content: center;
}

/* Style for the right side of the home page */
.home-page .right {
    width: 50%;
    background-color: white;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

/* Paragraph styling for the sliding sentences */
.sentence-section p {
    font-size: 40px;
    color: black;
    text-align: center;
    font-weight: bold;
}

/* Styling for the logo */
.nav-logo {
    font-family: 'Fredoka One', cursive; /* Fredoka One font */
    font-weight: 700; /* Bold weight */
    font-size: 3rem; /* Adjust size as needed */
    letter-spacing: -2px; /* Make letters closer */
    color: black; /* Default color for 'Gen' */
    text-transform: uppercase; /* All letters in uppercase */
    text-decoration: none;
    display: inline-block;
}

.nav-logo span {
    color: #ffd700; /* Color for 'Nex' */
    text-shadow: 3px 3px 6px rgba(0,0,0,0.2); /* Optional: Adds depth */
}

/* Styling for the buttons */
.home-page .right button {
    margin: 10px 20px;
    padding: 15px 30px;
    font-size: 1.5rem;
    border: none;
    cursor: pointer;
    background-color: black;
    width: 150px;
    text-align: center;
    transition: background-color 0.3s;
    color: #ffd700; /* Set text color to yellow */
}

.home-page .right button:hover {
    background-color: #ffd700;
    color: #333;
}

/* Remove underline from links inside buttons */
.home-page .right button a {
    text-decoration: none;
    color: #ffd700;
}

.home-page .right button a:hover {
    text-decoration: none;
    color: #333;
}

/* Styling for the footer/about section */
.home-page .right .footer-about {
    position: absolute;
    bottom: 10px;
    left: 10px;
    font-size: 16px;
    color: #059b16; /* Original color, will be overridden below for black color */
    font-weight: bold;
}

/* Link component styling */
.home-page .right .footer-about a,
.home-page .right .footer-about a:visited {
    text-decoration: none;
    color: black; /* Changed color to black */
}

.home-page .right .footer-about a:hover {
    color: #ffd700;
}

/* -- Additional styling for the footer/about section for alignment -- */
.home-page .right .footer-about {
    align-self: center; /* center of the flex container */
    width: 100%; /* Sets the width to 100% of the container */
    text-align: center; /* Centers the text within the footer-about */
    position: relative; /* Changes from absolute to relative positioning */
    font-size: 14px;
    color: black; /* Ensures the text is black */
    font-weight: bold;
    margin-top: auto; /* Pushes the footer-about to the bottom of the flex container */
    top: 250px; /* move About us downwards */
}

.nav-logo img {
    width: 50px; /* width of the logo*/
    height: auto; /* !! */
    display: block; /* !! */
    margin: 0 auto; /* Center the logo  */
    position: relative; /* Relative position */
    top: -40px; /* Move the logo 50px up from its original position */
}
