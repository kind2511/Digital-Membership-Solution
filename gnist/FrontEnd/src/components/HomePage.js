import React, { useEffect, useState } from 'react';
import { useAuth0 } from '@auth0/auth0-react';
import { Link } from 'react-router-dom';
import './HomePage.css';

const promoSentences = [
  "Bli medlem enkelt: Logg inn og start din reise hos oss",
  "Ditt nærvær teller - registrer besøket og nyt fordelene",
  "Stig opp i nivåer: Fra Noob til Pro, hver dag hos oss teller",
  "Hold deg oppdatert med dagens program - bli med på aktiviteten",
  "Din stemme teller: delta i meningsmålinger og bidra med forslag"
];

const HomePage = () => {
  const [displayedText, setDisplayedText] = useState('');
  const [sentenceIndex, setSentenceIndex] = useState(0);
  const { loginWithRedirect } = useAuth0();

  useEffect(() => {
    let currentTimer;

    const isCompleteSentence = displayedText === promoSentences[sentenceIndex];
    if (isCompleteSentence) {
      currentTimer = setTimeout(() => {
        setSentenceIndex((sentenceIndex + 1) % promoSentences.length);
        setDisplayedText('');
      }, 2000);
    } else {
      currentTimer = setTimeout(() => {
        setDisplayedText(promoSentences[sentenceIndex].substr(0, displayedText.length + 1));
      }, 50);
    }

    return () => {
      clearTimeout(currentTimer);
    };
  }, [displayedText, sentenceIndex]);

  const handleAuthAction = () => {
    loginWithRedirect();
  };

  return (
    <div className="home-page">
      <div className="left">
        <div className="sentence-section">
          <p>{displayedText}</p>
        </div>
      </div>
      <div className="right">
        <div className="nav-logo">Kom i<span> gang</span></div>
        <button onClick={handleAuthAction}>Login/Register</button>
      </div>
      <div className="kontakt-oss-link-container">
        <Link to="/kontakt-oss" className="kontakt-link">Kontakt oss</Link>
      </div>
    </div>
  );
};

export default HomePage;
