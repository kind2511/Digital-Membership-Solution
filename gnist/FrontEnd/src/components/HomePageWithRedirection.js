import React, { useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import { useAuth0 } from '@auth0/auth0-react';
import HomePage from './HomePage';
import Loading from './Loading';

function HomePageWithRedirection() {
  const { isAuthenticated, user, isLoading } = useAuth0();
  const navigate = useNavigate();

  useEffect(() => {
    if (!isLoading && isAuthenticated) {
    
      const userProfileComplete = false; 

      // Extract the role from the user's data
      const role = user['https://my-namespace/role'];

      // Redirect based on the user's role and profile completeness
      if (!userProfileComplete) {
        navigate('/user-info');
      } else if (role === 'admin') {
        navigate('/admin-dashboard');
      } else {
        navigate('/user-dashboard');
      }
    }
  }, [isAuthenticated, isLoading, navigate, user]);

  return isLoading ? <Loading /> : <HomePage />;
}

export default HomePageWithRedirection;
