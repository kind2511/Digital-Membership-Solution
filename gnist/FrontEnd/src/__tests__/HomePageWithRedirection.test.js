import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Auth0Context } from '@auth0/auth0-react';
import HomePageWithRedirection from '../components/HomePageWithRedirection';
import HomePage from '../components/HomePage';
import Loading from '../components/Loading';

// Mock child components
jest.mock('../components/HomePage', () => () => <div>HomePage Component</div>);
jest.mock('../components/Loading', () => () => <div>Loading Component</div>);

// Mock Auth0 context
const mockAuth0Context = {
  isAuthenticated: false,
  user: null,
  isLoading: true,
};

// Mock fetch
global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve({ registered: false }),
  })
);

test('HomePageWithRedirection component snapshot', () => {
  const component = renderer.create(
    <Auth0Context.Provider value={mockAuth0Context}>
      <MemoryRouter>
        <HomePageWithRedirection />
      </MemoryRouter>
    </Auth0Context.Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
