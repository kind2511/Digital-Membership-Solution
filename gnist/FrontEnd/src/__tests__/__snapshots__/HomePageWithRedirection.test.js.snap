// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageWithRedirection component snapshot 1`] = `
<div>
  Loading Component
</div>
`;
